spec/unit/puppet/provider/conf_setting/ruby_spec.rb

```diff
@@ -212,14 +212,106 @@ def validate_file(expected_content,tmpfile = tmpfile)
       validate_file(" test_key_1 : { setting1 : \"helloworld\" }\n", emptyfile)
     end
 
-    it "should be able to handle variables of any type" do
+    it "should be able to handle variables of boolean type" do
       resource = Puppet::Type::Hocon_setting.new(common_params.merge(
-          :setting => 'test_key_1.master', :value => true))
+          :setting => 'test_key_1.master', :value => false))
       provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(false)
+    end
+
+    it "should be able to handle variables of integer type" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => 12))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(12)
+    end
+
+    it "should be able to handle variables of float type" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => 12.24))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql(12.24)
+    end
+
+    it "should be able to handle arrays" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => [1, 2, 3, 4]))
+      provider = described_class.new(resource)
+      provider.create
       expect(provider.exists?).to be true
-      expect(provider.value).to eql(true)
+      expect(provider.value).to eql([1, 2, 3, 4])
     end
 
+    it "should be able to handle maps" do
+      resource = Puppet::Type::Hocon_setting.new(common_params.merge(
+                                                     :setting => 'test_key_1.master', :value => {"a" => 1, "b" => 2}))
+      provider = described_class.new(resource)
+      provider.create
+      expect(provider.exists?).to be true
+      expect(provider.value).to eql({"a" => 1, "b" => 2})
```

Comment:
Just to validate your logic here:

This test only tests that the value of the Puppet resource ends up being set to a map, right?  It doesn't actually test that the module can successfully pass a map into the HOCON API, and validate that we can read the same map back out of the HOCON API?  Do you have coverage for that somewhere else in the module?

I'm thinking you probably don't need to have a test for roundtripping a map value to a file, because we have tests for that in the gem, but it might still be worth just testing that the Puppet module is interacting with the HOCON gem API successfully.